### PR TITLE
ENG-8950: Use temp directory under voltdbroot to extract native lib f…

### PR DESCRIPTION
**Pull request by manjujames**
…rom jars.

### CODE REVIEW

File: lib/python/voltcli/runner.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def execute(self, java_class, java_opts_override, *args, **kwargs):
         java_opts = utility.merge_java_options(environment.java_opts, java_opts_override)
         java_args.extend(java_opts)
         java_args.append('-Dlog4j.configuration=file://%s' % os.environ['LOG4J_CONFIG_PATH'])
-	java_args.append('-Djava.library.path=dummyvalue')
+	java_args.append('-Djava.library.path=default')
```
Comment (akhanzode) [inline lib/python/voltcli/runner.py]: indent.

File: src/frontend/org/voltdb/EELibraryLoader.java [hunk 2]
```diff
@@ -122,7 +124,13 @@ private static File getNativeLibraryFile(String libname) {
             }
         }
 
-        File tmpFilePath = new File(System.getProperty(VOLT_TMP_DIR, System.getProperty("java.io.tmpdir")));
+        File tmpFilePath = getTempFilePath();
+        // set the temp dir for snappy also. Unless it is set already to override what we are doing here.
+        String currSnappyTempValue = System.getProperty(SNAPPY_TEMP_DIR_PROP);
+        if (currSnappyTempValue == null) {
+            System.setProperty(SNAPPY_TEMP_DIR_PROP, tmpFilePath.getAbsolutePath());
```
Comment (akhanzode) [inline src/frontend/org/voltdb/EELibraryLoader.java]: log what directory is getting used for storing temp jni libs

File: tests/frontend/org/voltdb/TestEELibraryLoader.java [hunk 3]
```diff
@@ -53,5 +56,24 @@ public void testLoader() {
...
+        when(cluster.getVoltroot())
+            .thenReturn(System.getProperty("java.io.tmpdir"));
+        when(catContext.getCluster())
+            .thenReturn(cluster);
+        when(mockitovolt.getCatalogContext())
+            .thenReturn(catContext);
+
+        VoltDB.replaceVoltDBInstanceForTest(mockitovolt);
+        VoltDB.initialize(configuration);
+        assertTrue(EELibraryLoader.loadExecutionEngineLibrary(true));
```
Comment (akhanzode) [inline tests/frontend/org/voltdb/TestEELibraryLoader.java]: Do u want to add test that starts a cluster and makes sure that JNI's (ours and snappy) are indeed under voltdbroot with correct name?

File: src/frontend/org/voltdb/EELibraryLoader.java [hunk 4]
```diff
@@ -22,13 +22,14 @@
...
 import java.io.InputStream;
 import java.util.UUID;
-import static org.voltcore.common.Constants.VOLT_TMP_DIR;
 
 import org.voltcore.logging.VoltLogger;
 
 public class EELibraryLoader {
 
     public static final String USE_JAVA_LIBRARY_PATH = "use.javalib";
+    public static final String LIB_TEMP_DIR = "temp/jnilib";
```
Comment (akhanzode) [inline src/frontend/org/voltdb/EELibraryLoader.java]: why temp/jnilib and why not just jnilib under voltdbroot?

Comment (manjujames) [inline src/frontend/org/voltdb/EELibraryLoader.java]: That's fine with me.

**Comment by manjujames** [comment]
Squash merged

**Pull request closed without merging**